src/Impostor.Api/Net/Manager/ICompatibilityManager.cs

```diff
@@ -0,0 +1,61 @@
+namespace Impostor.Api.Net.Manager
+{
+    /**
+     * <summary>
+     * Maintain an internal compatibility list of versions that are allowed to connect to the server, and which game
+     * versions they are allowed to play with.
+     * </summary>
+     */
+    public interface ICompatibilityManager
+    {
+        public enum VersionCompareResult
+        {
+            Compatible,
+            ClientTooOld,
+            ServerTooOld,
+            Unknown,
+        }
+
+        /**
+         * <summary>
+         * Get the compatibility group of a certain client version. If two versions are in the same compatibility group,
+         * they can play a game together without issues.
+         * </summary>
+         * <param name="clientVersion">The client version to check for.</param>
+         * <param name="compatGroup">If the result is Compatible, the compat group.</param>
+         * <returns>
+         * Whether this version is supported by the server at the moment and if not, whether it is too old or too new.
+         * </returns>
+         */
+        public VersionCompareResult TryGetCompatibilityGroup(int clientVersion, out int? compatGroup);
```

Comment:
We only ever use this to either get the compare result or the compat group, shouldn't this be split into 2 methods - Compare and GetCompatiblityGroup? Eventually only use this inside `ClientManager` and pass the compat group down to `Game` inside `IClient`

Reply:
Splitted this into two methods:
- CanConnectToServer compares one client to what the server supports
- CanJoinGame compares two clients to each other to see if they can join a game together.
Original method is poof unless someone has a good usecase for it